packages/pipeline-editor/src/PipelineController/index.ts

```diff
@@ -444,10 +445,44 @@ class PipelineController extends CanvasController {
         nodes.push(...c.node_types);
       }
     }
+    nodes = this.propagatePipelineDefaultProperties(nodes, this.palette);
 
     return nodes;
   }
 
+  propagatePipelineDefaultProperties(
+    nodes: NodeType[],
+    palette: PaletteV3
+  ): NodeType[] {
+    return produce(nodes, (draft: any) => {
+      for (const prop of palette.properties?.uihints?.parameter_info ?? []) {
+        const propValue = this.getPipelineFlow()?.pipelines?.[0]?.app_data
+          ?.properties?.pipeline_defaults?.[
+          prop.parameter_ref.replace(/^elyra_/, "")
+        ];
+
+        draft.forEach((node: any) => {
+          const propIndex = node.app_data.properties.uihints.parameter_info.findIndex(
```

Comment:
Maybe we should check if `propIndex` is valid before using it to get the `nodeProp` below.

Reply:
that's how it was done in the code I copied this from, but I figured that checking if propIndex was valid was redundant when I could get check if nodeProp was valid. Also it would be less code since with an invalid propIndex we must check it's value but checking nodeProp only needs to check for existence